packages/ui/v2/core/apps/AllApps.tsx

```diff
@@ -0,0 +1,109 @@
+import { useAutoAnimate } from "@formkit/auto-animate/react";
+import type { Credential } from "@prisma/client";
+import { UIEvent, useEffect, useRef, useState } from "react";
+import { ChevronLeft, ChevronRight } from "react-feather";
+
+import { classNames } from "@calcom/lib";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import type { App } from "@calcom/types/App";
+
+import AppCard from "./AppCard";
+
+type AllAppsPropsType = { apps: (App & { credentials: Credential[] | undefined })[] };
+
+export default function AllApps({ apps }: AllAppsPropsType) {
+  const { t } = useLocale();
+  const [selectedCategory, setSelectedCategory] = useState<string | null>(null);
+  const [appsContainerRef] = useAutoAnimate<HTMLDivElement>();
+  const appCategoriesListRef = useRef<HTMLUListElement>(null);
+  const [appCategoriesShowArrowScroll, setAppCategoriesShowArrowScroll] = useState({
+    left: false,
+    right: false,
+  });
+
+  useEffect(() => {
+    const appCategoryList = appCategoriesListRef.current;
+    if (appCategoryList && appCategoryList.scrollWidth > appCategoryList.clientWidth) {
+      setAppCategoriesShowArrowScroll({ left: false, right: true });
+    }
+  }, []);
+
+  const calculateScroll = (e: UIEvent<HTMLUListElement>) => {
+    setAppCategoriesShowArrowScroll({
+      left: e.currentTarget.scrollLeft > 0,
+      right:
+        Math.floor(e.currentTarget.scrollWidth) - Math.floor(e.currentTarget.offsetWidth) !==
+        Math.floor(e.currentTarget.scrollLeft),
+    });
+  };
```

Comment:
Good idea! Should it be located at `packages/lib/hooks` to be able to import it from `@calcom/lib/hooks/useShouldShowArrows`? It seems the hooks in there are very high level, so perhaps it would be better to be located in `@calcom/ui/v2/hooks`?